content/blog/announcing-pulumi-preview-improvments/index.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Announcing Pulumi Preview Improvements"
+date: 2020-10-27
+draft: false
+meta_desc: "Announcing a significant improvement to the pulumi preview experience."
+#meta_image: 
+authors:
+    - paul-stack
+tags:
+    - enhancement
+    - pulumi
+---
+
+Today we are announcing a minor but significant improvement to the Pulumi [preview]({{< relref "/docs/reference/cli/pulumi_preview" >}}
+experience.
+
+<!--more-->
+
+## Change in Behaviour
+
+We can look at a sample Pulumi application that creates a Google GKE cluster. Pulumi then deploys an application into the
+Kubernetes cluster.
+
+{{< gist pulumipus 0c6e4ede557600e114885869ea0aeb96 >}}
+
+A simple change to the labels for the GKE cluster led to a misleading preview experience that told users that the Kubernetes
+resources were going to be replaced:
```

Comment:
We should try to frame this differently.  Explain why this previous behaviour was the way it was, not frame it as "bad", but frame the opportunity to make it better.  

Something like:
1. Here's what you previously saw during the preview.
2. Note that the actual update would not need to replace any of those resources
3. This is because Pulumi was trying to be conservatively correct - let you know everything that *could* happen
4. It thought that the update of the Cluster could potentially lead to outputs of the cluster changing which other resources depend on.
5. In practice though, that won't ever happen here.  But Pulumi previously didn't have a way to check for that, and so had to be conservative to be correct.

And then note later what changed.
1. Pulumi now supports asking the individual resource providers about what outputs will change as part of an update to a resource.  
2. This allows the providers (like the GCP provider in this case) to give more specific information, which allows Pulumi to provide more accurate previews.

(Would be good for @pgavlin to review what I wrote there - but I think if we are going to do a post here we likely want a narrative along those lines).

Reply:
I really like this - this sums it up way nicer - thank you!

Reply:
> Pulumi now supports asking the individual resource providers about what outputs will change as part of an update to a resource.

"how outputs will change" more accurately describes what the changes do, as we also include information about specific values.

>  This allows the providers (like the GCP provider in this case) to give more specific information, which allows Pulumi to provide more accurate previews.

👍

Reply:
The description of our prior behavior is spot on.